apps/builder/src/features/user/components/UserPreferencesForm.tsx

```diff
@@ -49,7 +49,7 @@ export const UserPreferencesForm = () => {
   };
 
   const updateLocale = (locale: keyof typeof localeHumanReadable) => () => {
-    document.cookie = `NEXT_LOCALE=${locale}; path=/; max-age=31536000`;
+    updateUser({ preferredLanguage: locale });
```

Comment:
```suggestion
    document.cookie = `NEXT_LOCALE=${locale}; path=/; max-age=31536000`;
    updateUser({ preferredLanguage: locale });
```

Better to keep the cookie mechanism to make sure all the internal are working as expected with Next.js